pagopa/io-app
# feat(Firma con IO): [SFEQS-1163] Add user data sharing screen

## Short description
In this PR it has been added a screen to inform the user of the data that will be shared with the applicant.

![Simulator Screen Shot - iPhone 13 - 2022-12-01 at 10 26 54](https://user-images.githubusercontent.com/49342144/205017456-b22b9cd1-9f86-44ce-b4db-c101f41c7f88.png)

## List of changes proposed in this pull request
- Added `FciDataSharingScreen`
- Updated `FciStackNavigator`
- Update `backendFci` with request to GET QTSP metadata and POST `document_url` to be filled
- Added the related networking saga
- Update `watchFciSaga` to handle new requests
- Added some tests

## How to test
* if not yet merged get this branch https://github.com/pagopa/io-dev-api-server/tree/SFEQS-1082-fci-endpoints from io-dev-server-api
* enable FCI_ENABLED ff
* tap on a right signature message with label WAIT_FOR_SIGNATURE
* tap on CTA of the message details
* show document and check required signatureFields
